dandi/dandiapi.py

```diff
@@ -554,13 +554,63 @@ def get_dandiset(
                     return d.for_version(version_id)
             return d
 
-    def get_dandisets(self) -> Iterator[RemoteDandiset]:
+    def get_dandisets(
+        self,
+        *,
+        draft: bool | None = None,
+        embargoed: bool | None = None,
+        empty: bool | None = None,
+        mine: bool | None = None,
+        order: str | None = None,
+        search: str | None = None,
+    ) -> Iterator[RemoteDandiset]:
         """
         Returns a generator of all Dandisets on the server.  For each Dandiset,
         the `RemoteDandiset`'s version is set to the most recent published
         version if there is one, otherwise to the draft version.
-        """
-        for data in self.paginate("/dandisets/"):
+
+        .. versionchanged:: 0.61.0
+
+            ``draft``, ``embargoed``, ``empty``, ``mine``, ``order``, and
+            ``search`` parameters added
+
+        :param draft:
+            If true, Dandisets that have only draft versions (i.e., that
+            haven't yet been published) will be included in the results
+            (default true)
+
+        :param embargoed:
+            If true, embargoed Dandisets will be included in the results
+            (default false)
+
+        :param empty:
+            If true, empty Dandisets will be included in the results (default
+            true)
+
+        :param mine:
+            If true, only Dandisets owned by the authenticated user will be
+            retrieved (default false)
+
+        :param order:
+            The field to sort the results by.  The accepted field names are
+            ``"id"``, ``"name"``, ``"modified"``, and ``"size"``.  Prepend a
+            hyphen to the field name to reverse the sort order.
+
+        :param search:
+            A search string to filter the returned Dandisets by.  The string is
+            searched for in the metadata of Dandiset versions.
+        """
+        for data in self.paginate(
+            "/dandisets/",
+            params={
+                "draft": draft,
+                "embargoed": embargoed,
+                "empty": empty,
+                "ordering": order,
+                "search": search,
+                "user": "me" if mine else None,
```

Comment:
should we stick to API's interface here and expose as `user`?  May be admins (later?) might be able to query for some other users?

Reply:
No, I don't think having a parameter that can only take one value makes much sense.

Reply:
I agree... as we are to RF order vs ordering, please file another issue on this particular one. May be indeed the intent would be to expose querying for other users as well. If not -- we better recommend to refactor it to `mine`.

Reply:
Issue filed: https://github.com/dandi/dandi-archive/issues/1879